src/CoreWCF.Primitives/src/CoreWCF/Security/DataProtectionSecurityStateEncoder.cs

```diff
@@ -3,64 +3,26 @@
 
 using System;
 using System.Security.Cryptography;
-using System.Text;
-using CoreWCF.Runtime;
+using Microsoft.AspNetCore.DataProtection;
 
 namespace CoreWCF.Security
 {
     public class DataProtectionSecurityStateEncoder : SecurityStateEncoder
     {
-        private readonly byte[] _entropy;
+        private readonly Lazy<IDataProtector> _dataProtector;
 
-        public DataProtectionSecurityStateEncoder() : this(true)
-        {
-            // empty
-        }
-
-        public DataProtectionSecurityStateEncoder(bool useCurrentUserProtectionScope) : this(useCurrentUserProtectionScope, null)
-        { }
-
-        public DataProtectionSecurityStateEncoder(bool useCurrentUserProtectionScope, byte[] entropy)
-        {
```

Comment:
Removing a public api is a breaking change and would need moving to a 2.0 release. Even though the data is called entropy, it wasn't actually used as Entropy by ProtectedData, it was just a way to make the encryption unique and prevent other code running on the same machine/user from being able to decode data encoded in other places. There are some security attacks which target that scenario. This entropy acts in a similar way to what is passed to DataProtector as a purpose. I think we can support this api by converting the byte array into a base64 encoded string and using that as the purpose.

Reply:
i reverted to ensure there is no breaking changes in public api